pkg/reconciler/kubernetes/adapter.go

```diff
@@ -213,6 +213,18 @@ func (g *kubeClientAdapter) newNamespaceUnstruct(namespace string) (*unstructure
 	return nsUnstructs[0], nil
 }
 
+func (g *kubeClientAdapter) DeleteResourceByKindAndNameAndNamespace(kind, name, namespace string) (*Resource, error) {
```

Comment:
If all parameters are required, maybe we call it just `DeleteResource` (quite long fct name otherwise :) ) ?

Reply:
Done